docs/installation/viam-micro-server-dev.md

```diff
@@ -20,275 +20,262 @@ aliases:
-
-   Run the following command to install `cargo-generate`:
+1. Install `espup` and ESP development utilities with `cargo`:
 
    ```sh { class="command-line" data-prompt="$"}
-   cargo install cargo-generate
+   cargo install cargo-espflash cargo-generate espflash espup ldproxy
    ```
 
-4. Install `espflash`
-
-   Run the following command to install `espflash`:
+1. Use `espup` to download and install the ESP Rust toolchain:
 
    ```sh { class="command-line" data-prompt="$"}
-   cargo install espflash
+   espup install -s -f /dev/null -v 1.76.0
    ```
 
-## Set up your development environment
+{{< alert title="Note" color="tip" >}}
+Prior versions of the above `espup` instructions created a file called `export-rs.sh` which needed to be sourced before proceeding.
+That requirement [no longer applies](https://github.com/esp-rs/esp-idf-hal/issues/319#issuecomment-1785168921) for `viam-micro-server` development.
+The above command redirects `espup`'s production of the setup script to `/dev/null` to avoid cluttering the home directory.
+If you would like to instead retain the setup script, replace `/dev/null` in the above command with the location where you would like the script to be written, or remove the `-f /dev/null` entirely and the file will be written to `$HOME/export-esp.sh` by default.
+{{< /alert >}}
 
-{{< tabs >}}
-{{% tab name="Canon CLI (recommended)" %}}
+## Development tasks
 
-[Canon](https://github.com/viamrobotics/canon) is a CLI utility for managing a Docker-based canonical environment.
+There are three main types of development tasks relevant for `viam-micro-server`:
 
-1. [Install Docker Engine](https://docs.docker.com/engine/install/). Canon requires a working installation of Docker Engine.
-2. If you are running Docker Engine on Linux, make sure that you go through the [post installation steps](https://docs.docker.com/engine/install/linux-postinstall/) to run Docker automatically on startup.
+- Creating a project that combines `viam-micro-server` with one or more modules to produce a new firmware instance with expanded functionality.
+- Creating modules that can integrate with `viam-micro-server` to deliver new opt-in functionality or device support.
+- Working on `viam-micro-server` itself, to add features and fix bugs, or to produce a build for a specific ESP-IDF version or platform for which Viam does not offer a pre-built solution.
 
 {{< tabs >}}
-{{% tab name="Direct Go Install (Linux)" %}}
+{{% tab name="Create a New Project" %}}
 
-3. Run `go version` to confirm your system has [Go 1.19](https://golangtutorial.dev/news/go-1.19-version-released/#major-changes-in-go-119-version) or later installed.
-4. Install canon:
+1.  Create a new machine and obtain its credentials
 
-   ```sh { class="command-line" data-prompt="$"}
-   go install github.com/viamrobotics/canon@latest
-   ```
+- Navigate to [the Viam app](https://app.viam.com) and [add a new machine](/cloud/machines/#add-a-new-machine) in your desired location.
```

Comment:
these aren't indented, I think there's a bug with that and that's probably why? Maybe just use regular text & paragraphs.

Reply:
changed